### PR TITLE
WIP: Destroying windows when they are closed

### LINKED ISSUE #224: Windows are not destroyed properly?
Seems that any window Pext opens is just hidden instead of being destroyed? And on Windows, the peek feature shows all these:

image

- OS: Windows 10
- Pext v0.26-27-g044fc99

### PR DESCRIPTION
**Pull request by TangentFoxy**
Fixes #224

### CODE REVIEW

**Comment by TangentFoxy** [comment]
Progress on QML files:

- [x] AboutDialog
- [x] ChoiceDialog (slightly uncertain)
- [x] CommandArgsDialog (slightly uncertain)
- [x] ConfirmQuitDialog
- [ ] CriticalErrorDialog
- [x] InputRequestDialog (slightly uncertain)
- [x] InstallFromRepositoryDialog (slightly uncertain)
- [ ] InstallModuleFromRepositoryNoModulesAvailableDialog
- [x] InstallModuleFromRepositorySelectModuleDialog (slightly uncertain)
- [x] InstallModuleFromURLDialog (very uncertain)
- [x] InstallThemeFromURLDialog (very uncertain)
- [x] LoadModuleDialog (slightly uncertain)
- [x] LoadProfileDialog (slightly uncertain)
- [x] LoadThemeDialog (slightly uncertain)
- [ ] main (does not need to be modified?)
- [x] ManageDialog
- [x] ManageProfilesDialog
- [ ] ModuleData
- [ ] NoModulesInstalledDialog
- [ ] NoThemesInstalledDialog
- [ ] OnlyOneProfileDialog
- [x] QuestionDialog (slightly uncertain)
- [x] RenameProfileDialog (slightly uncertain)

* **(slightly uncertain)**: I think they are fine being destroyed where I put the `destroy()` calls but I am unsure. Each of these is calling something (synchronously?) right before being destroyed.
* **(very uncertain)**: These are making an HTTP request right before where I added `destroy()` so I am unsure if this is breaking those requests.

**Comment by TangentFoxy** [comment]
At this point I've done all I can on this without doing some local testing (I know how I can test this, i just didn't know how to test AppVeyor building things right on that other issue).

I also possibly need feedback on what the heck ModuleData and the polyfill JS file are doing in order to see if I need to change anything with either of them. (I'm guessing I don't...but unsure.)

**Comment by TheLastProject** [comment]
ModuleData contains the actual mode and is part of the main screen and shouldn't be destroyed. The polyfill JS is to ensure Pext also works on older Qt versions which have an older JavaScript engine. The JS isn't a window so doesn't need to be destroyed either, it's just a JavaScript file.

File: pext/qml/InstallModuleFromURLDialog.qml [hunk 1]
```diff
@@ -65,6 +65,9 @@ Dialog {
 
         xmlhttp.open("GET", textfield.text, true);
         xmlhttp.send();
+
+        destroy();
```
Comment (TheLastProject) [inline pext/qml/InstallModuleFromURLDialog.qml]: Please put this destroy inside the onreadystatechange after installRequest is called

File: pext/qml/InstallThemeFromURLDialog.qml [hunk 2]
```diff
@@ -60,6 +60,9 @@ Dialog {
 
         xmlhttp.open("GET", textfield.text, true);
         xmlhttp.send();
+
+        destroy();
```
Comment (TheLastProject) [inline pext/qml/InstallThemeFromURLDialog.qml]: Please put this destroy inside the onreadystatechange after installRequest is called

File: pext/qml/InstallModuleFromURLDialog.qml (hunk 1 above)
Comment (TangentFoxy) [reply pext/qml/InstallModuleFromURLDialog.qml]: Commited! :D

File: pext/qml/InstallThemeFromURLDialog.qml (hunk 2 above)
Comment (TangentFoxy) [reply pext/qml/InstallThemeFromURLDialog.qml]: Commited! :D

**Comment by TheLastProject** [comment]
Looking quite good. Did you manage to test yet if this fixes the issue?

**Comment by TangentFoxy** [comment]
Not yet. Trying a few different things as I go:

- [x] Xubuntu in a VM (failed due to error installing PyOpenGL_accelerate)
- [ ] Majaro in a VM (doing this now)
- [ ] Windows 10 (my native current OS)
- [ ] Windows 10 in a VM (maybe?)

**Comment by TheLastProject** [comment]
Just merging this because want to do a release soon and it at least doesn't seem to break anything. Please get back to me if you can confirm it also fixes stuff.

**Pull request merged**